app/views/rankings/index.html.erb

```diff
@@ -1,6 +1,10 @@
 <h1><%= ranking_title(params[:time_period]) %></h1>
 <p><%= render partial: 'shared/ranking_links' %></p>
-<div class="leaderboard_list">
-  <%= render :partial => 'ranking', :locals => {:leaderboard => @leaderboard} %>
-</div>
-<p><strong>Note:</strong> leaderboard updates every <%= distance_of_time_in_words(@cache_expiration) %>.</p>
\ No newline at end of file
+
+<% cache [:user_ranking_partial, params[:time_period]] do %>
+  <%= render :partial => 'ranking', :locals => { :leaderboard => @leaderboard } %>
+<% end %>
```

Comment:
I put the fragment cache to avoid N+1 queries on the user relation.
I don't consider this a definitive implementation about leaderboard, but only a temporary fix to make leaderboards working.

Btw, I forgot to put the `params[:page]` value into the cache key.